lib/ddtrace/test_tracer.rb

```diff
@@ -0,0 +1,22 @@
+require 'ddtrace/tracer'
+require 'concurrent'
+
+module Datadog
+  class TestTracer < Tracer
```

Comment:
is this `TestTracer` supposed to be used only for tests? Because in that case we already have a `helper.rb` module that offers a `get_test_tracer` function that returns a tracer with a fake `Writer` (ref: https://github.com/nerdrew/dd-trace-rb/blob/bcf0fa336ff33b1a647587ed9dce3b27eee441f3/test/helper.rb#L32-L35)

If it doesn't fit your need, we may thing to refactor better the `helper.rb` module so that the test tracer is extensible. What do you think?

Reply:
Yes, the test tracer (`FauxTransport` and `SpyTransport`) in `test/helper.rb` is what I want. Can we extract all those classes to something re-usable so I can use them when testing my app without pulling in `minitest` and `minitest/autorun`? I think that would satisfy my needs.

Are these considered part of the public api for the gem (though not intended for production use, only testing apps)?

Reply:
As far as I can tell the helper isn't available in the Gem, only the source code.

Reply:
Yeah, the helper would only be available in our own test suite for `ddtrace`, not in test suites for applications that implement `ddtrace`.

@nerdrew, I like the idea of having a`Test` namespace in which we can 'first class' some of these common test components. Then anyone could use the same test components to drive their own tests.

Concerning the RSpec `expect { .. }.to trace` idea, I think this is wonderful; I think we would find a lot of use for this in our test suite. However, since not everyone uses RSpec, we might want to be careful about how we add test framework specific features. I think this would warrant further discussion about whether such a feature belongs in this gem, or in a separate tracer testing gem (as is common for RSpec extensions.)